cli/util/util.go

```diff
@@ -399,3 +403,188 @@ func IsRegularFile(filePath string) bool {
 
 	return fileInfo.Mode().IsRegular()
 }
+
+// IsDeprecated checks wethere version of program is below 1.10.0.
+func IsDeprecated(version string) bool {
+	splitedVersion := strings.Split(version, ".")
+	if len(splitedVersion) < 2 {
+		return false
+	}
+	if splitedVersion[0] == "1" && len(splitedVersion[1]) < 2 {
+		return true
+	}
+	return false
+}
+
+// CopyFilePreverse copies file from source to destination with perms.
+func CopyFilePreverse(src string, dst string) error {
+	// Read all content of src to data.
+	info, err := os.Stat(src)
+	if err != nil {
+		return err
+	}
+	data, err := ioutil.ReadFile(src)
+	if err != nil {
+		return err
+	}
+	// Write data to dst.
+	err = ioutil.WriteFile(dst, data, info.Mode().Perm())
+	return err
+}
+
+// CopyFileChangePerms copies file from source to destination with changing perms.
+func CopyFileChangePerms(src string, dst string, perms int) error {
+	// Read all content of src to data.
+	_, err := os.Stat(src)
+	if err != nil {
+		return err
+	}
+	data, err := ioutil.ReadFile(src)
+	if err != nil {
+		return err
+	}
+	// Write data to dst.
+	err = ioutil.WriteFile(dst, data, fs.FileMode(perms))
+	return err
+}
+
+// CopyDir copies the content of src to dst, src should be a full path.
+func CopyDir(src, dst string) error {
+
+	return filepath.Walk(src, func(path string, info fs.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+
+		// copy to this path
+		outpath := filepath.Join(dst, strings.TrimPrefix(path, src))
+
+		if info.IsDir() {
+			os.MkdirAll(outpath, info.Mode())
+			return nil
+		}
+
+		// handle irregular files
+		if !info.Mode().IsRegular() {
+			switch info.Mode().Type() & os.ModeType {
+			case os.ModeSymlink:
+				link, err := os.Readlink(path)
+				if err != nil {
+					return err
+				}
+				return os.Symlink(link, outpath)
+			}
+			return nil
+		}
+
+		in, _ := os.Open(path)
+		if err != nil {
+			return err
+		}
+		defer in.Close()
+
+		fh, err := os.Create(outpath)
+		if err != nil {
+			return err
+		}
+		defer fh.Close()
+
+		fh.Chmod(info.Mode())
+
+		_, err = io.Copy(fh, in)
+		return err
+	})
+}
+
+// ResolveSymlink resolves symlink path.
+func ResolveSymlink(linkPath string) (string, error) {
+	resolvedLink, err := os.Readlink(linkPath)
+	if err != nil {
+		return "", err
+	}
+	// Output of os.Readlink is OS-dependent, so need to check if path is absolute.
+	if !filepath.IsAbs(resolvedLink) {
+		resolvedLink = path.Join(path.Dir(linkPath), resolvedLink)
+	}
+	return resolvedLink, nil
+}
+
+// RunCommandAndGetOutput returns output of command.
+func RunCommandAndGetOutput(program string, args ...string) (string, error) {
+	out, err := exec.Command(program, args...).Output()
+	if err != nil {
+		return "", err
+	}
+
+	return strings.TrimSpace(string(out)), nil
+}
+
+// ExtractTar extracts tar archive.
+func ExtractTar(tarName string) error {
```

Comment:
Do we really need to implement this by ourselves? I use the extract package for tgz extraction in "create" command implementation. Pls, check if it is applicable for your purpose.

Reply:
We had a conversation with @LeonidVas that if we can write something with our hands, we don't need to use additional dependencies. That function was in our code since 'tt coredump', so I don't want to use external. Also, I encountered some problems with using that package in 'tt coredump unpack' it was'nt working properly with some archives.

Reply:
"if we can write something with our hands, we don't need to use additional dependencies" is not true for everything that we can write. This is coredump issue that it have its own unpack implementation. I don't want to have this code in common util package, because we already have "extract" available.